src/hooks/useSyncChainQuery.ts

```diff
@@ -31,8 +32,19 @@ export default function useSyncChainQuery() {
   const [searchParams, setSearchParams] = useSearchParams()
 
   useEffect(() => {
-    if (isActive && urlChainId && chainId !== urlChainId) {
+    // Change a user's chain on pageload of the connected chainId does not match the query param chain
+    if (isActive && urlChainId && chainIdRef.current === chainId && chainId !== urlChainId) {
       selectChain(urlChainId)
     }
-  }, [urlChainId, selectChain, searchParams, setSearchParams, isActive, chainId])
+    // If a user has a connected wallet and has manually changed their chain, delete the chain query param
+    else if (account && urlChainId && chainIdRef.current !== chainId && chainId !== urlChainId) {
+      searchParams.delete('chain')
```

Comment:
Why do we delete the chain param here, but set it through useSelectChain?

Reply:
useSelectChain is used when a user selects a chain from the in-app Network Switcher and sets the corresponding query parameter

This case is for when a user switches chains using their wallet so we want to clear the existing query parameter if it exists

Reply:
why do we want to clear it? why do we have two separate ux's here?

Reply:
To clear the now out of date original query parameter